Sources/SwiftGD/RasterFormat.swift

```diff
@@ -0,0 +1,337 @@
+}
+
+/// Defines a formatter to be used on both, import & export, of one of libgd raster format with **none**-compressable export conversions
+public typealias LibGdRasterFormatter = LibGdImportableRasterFormatter & LibGdExportableRasterFormatter
+
+/// Defines a formatter to be used on both, import & export, of one of libgd built-in raster format with **compressable** export conversions
+public typealias LibGdCompressableRasterFormatter = LibGdImportableRasterFormatter & LibGdCompressableExportRasterFormatter
+
+// MARK: - Common Functions
+
+extension Quality {
+
+    /// libgd uses -1 as "default quality indicator"
+    /// Reference: https://libgd.github.io/manuals/2.2.4/files/gd_jpeg-c.html
+    public static let `default`: Quality = -1
+}
+
+extension LibGdImportableRasterFormatter {
+
+    /// Creates a `gdImagePtr` from given image data.
+    ///
+    /// - Parameter data: The image data of which an image should be instantiated.
+    /// - Returns: The `gdImagePtr` of the instantiated image.
+    /// - Throws: `Error` if import failed.
+    public func imagePtr(of data: Data) throws -> gdImagePtr {
+        let (pointer, size) = try data.memory()
+        guard let imagePtr = importFunction(size, pointer) else {
+            throw Error.invalidRasterFormat
+        }
+        return imagePtr
+    }
+}
+
+extension LibGdExportableRasterFormatter {
+
+    /// Creates a data representation of given `gdImagePtr`.
+    ///
+    /// - Parameter imagePtr: The `gdImagePtr` of which a data representation should be instantiated.
+    /// - Returns: The (raw) `Data` of the image
+    /// - Throws: `Error` if export failed.
+    public func data(of imagePtr: gdImagePtr) throws -> Data {
+        var size: Int32 = 0
+        guard let bytesPtr = exportFunction(imagePtr, &size) else {
+            throw Error.invalidRasterFormat
+        }
+        return Data(bytes: bytesPtr, count: Int(size))
+    }
+}
+
+extension LibGdCompressableExportRasterFormatter {
+
+    /// Creates a data representation of given `gdImagePtr`.
+    ///
+    /// - Parameter imagePtr: The `gdImagePtr` of which a data representation should be instantiated.
+    /// - Returns: The (raw) `Data` of the image
+    /// - Throws: `Error` if export failed.
+    public func data(of imagePtr: gdImagePtr) throws -> Data {
+        var size: Int32 = 0
+        let quality = min(max(self.quality, -1), 100)
+        guard let bytesPtr = exportFunction(imagePtr, &size, quality) else {
+            throw Error.invalidRasterFormat
+        }
+        return Data(bytes: bytesPtr, count: Int(size))
+    }
+}
+
+// MARK: - Concrete LibGd RasterFormatter
+
+/// Defines a formatter to be used on BMP import & export conversions
+public struct BMPRasterFormatter: LibGdCompressableRasterFormatter {
+
+    /// The image quality to apply on export operations
+    public var quality: Quality = .default
```

Comment:
? BMP supports compression, but I think it's either "use RLE or don't" – I'm not recall there being any quality aspect.